test/tracer.js

```diff
@@ -0,0 +1,183 @@
+        let startTime = Utils.getTimestampMicros();
+
+        let span = tracer.startSpan({
+            operationName: 'test-name',
+            childOf: context,
+            references: [new opentracing.Reference(opentracing.REFERENCE_CHILD_OF, context)],
+            startTime, startTime
+        });
+
+        assert.isOk(bufferEqual(span.context().traceId, traceId));
+        assert.isOk(bufferEqual(span.context().parentId, spanId));
+        assert.equal(span.context().flags, constants.SAMPLED_MASK);
+        assert.equal(span._start, startTime);
+    });
+
+    it ('start a child span represented as same span (Zipkins one-span-per-rpc)', () => {
+        let traceId = Utils.encodeInt64(1);
+        let spanId = Utils.encodeInt64(2);
+        let parentId = Utils.encodeInt64(3);
+        let flags = 1;
+        let context = new SpanContext(traceId, spanId, parentId, flags);
+        let startTime = Utils.getTimestampMicros();
+
+        let tags = {};
+        tags[`${opentracing_tags.SPAN_KIND}`] = opentracing_tags.SPAN_KIND_RPC_SERVER;
+        let span = tracer.startSpan({
+            operationName: 'test-name',
+            childOf: context,
+            references: [new opentracing.Reference(opentracing.REFERENCE_CHILD_OF, context)],
+            startTime: startTime,
+            tags
+        });
+
+        assert.isOk(bufferEqual(span.context().traceId, traceId));
+        assert.isOk(bufferEqual(span.context().parentId, parentId));
+        assert.equal(span.context().flags, constants.SAMPLED_MASK);
+        assert.equal(span._start, startTime);
+    });
+
+    it ('inject plain text headers into carrier, and extract span context with the same value', () => {
+        let savedContext = new SpanContext(
+            Utils.encodeInt64(1),
+            Utils.encodeInt64(2),
+            Utils.encodeInt64(3),
+            constants.SAMPLED_MASK
+        );
+        let keyOne = 'keyOne';
+        let keyTwo = 'keyTwo';
+        savedContext.setBaggageItem(keyOne, 'leela');
+        savedContext.setBaggageItem(keyTwo, 'bender');
+
+        let assertByFormat = (format) => {
+            let carrier = {};
+            tracer.inject(savedContext, format, carrier);
+            let extractedContext = tracer.extract(format, carrier);
+
+            assert.isOk(bufferEqual(savedContext.traceId, extractedContext.traceId));
+            assert.isOk(bufferEqual(savedContext.spanId, extractedContext.spanId));
+            assert.isOk(bufferEqual(savedContext.parentId, extractedContext.parentId));
+            assert.equal(savedContext.flags, extractedContext.flags);
+            assert.equal(savedContext.getBaggageItem(keyOne), extractedContext.getBaggageItem(keyOne));
+            assert.equal(savedContext.getBaggageItem(keyTwo), extractedContext.getBaggageItem(keyTwo));
+        }
+
+        assertByFormat(opentracing.FORMAT_TEXT_MAP);
+        assertByFormat(opentracing.FORMAT_HTTP_HEADERS);
```

Comment:
http headers codec must be url-encoding values, so the test should be slightly different (or have an extra one)